lib/src/collection/lru_map.dart

```diff
@@ -284,10 +285,10 @@ class LinkedLruHashMap<K, V> implements LruMap<K, V> {
   }
 
   @override
-  V update(K key, V update(V value), {V ifAbsent()}) {
+  V update(K key, V update(V value), {V ifAbsent()?}) {
     V newValue;
     if (containsKey(key)) {
-      newValue = update(this[key]);
+      newValue = update(this[key]!);
```

Comment:
Looks like non-nullness of `this[key]` implies `containsKey(key) == true`, so we could avoid the `!` like this:

```dart
V newValue;
final V? currentValue = this[key];
if (currentValue != null) {
  newValue = update(currentValue);
} else {
  ...
```

Reply:
Now that you need to declare nullability of the key type explicitly, we should probably allow null keys, since any use of them is much more likely to be intentional. I should probably make a similar change here as I did with BiMap and support this. Will probably do a separate PR for that though so we can discuss independently of NNBD migration.